File: library/src/com/sromku/simple/storage/helpers/OrderType.java
package com.sromku.simple.storage.helpers;

import java.io.File;
import java.util.Comparator;

public enum OrderType {

	NAME,
	/**
	 * Last modified is the first
	 */
	DATE,
	/**
	 * Smaller size will be in the first place
	 */
	SIZE;

	public Comparator<File> getComparator() {
		switch (ordinal()) {
		case 0: // name
			return new Comparator<File>() {
				@Override
				public int compare(File lhs, File rhs) {
					return lhs.getName().compareTo(rhs.getName());
				}
			}; 
		case 1: // date
			return new Comparator<File>() {
				@Override
				public int compare(File lhs, File rhs) {
					return (int) (rhs.lastModified() - lhs.lastModified());
				}
			}; 
		case 2: // size
			return new Comparator<File>() {
				@Override
				public int compare(File lhs, File rhs) {
					return (int) (lhs.length() - rhs.length());
				}
			}; 
		default:
			break;
		}
		return null;
	}
}

File: storage/src/main/java/com/snatik/storage/helpers/OrderType.kt
package com.snatik.storage.helpers

import java.io.File
import java.util.Comparator

enum class OrderType {

    NAME,
    /**
     * Last modified is the first
     */
    DATE,
    /**
     * Smaller size will be in the first place
     */
    SIZE;

    // name
    // date
    // size
    val comparator: Comparator<File>?
        get() {
            when (ordinal) {
                0 -> return comparatorForName()
                1 -> return comparatorForDate()
                2 -> return comparatorForSize()
                else -> {
                }
            }
            return null
        }

    private fun comparatorForName(): Comparator<File> {
        return Comparator { lhs, rhs -> lhs.name.compareTo(rhs.name) }
    }

    private fun comparatorForDate(): Comparator<File> {
        return Comparator { lhs, rhs -> (rhs.lastModified() - lhs.lastModified()).toInt() }
    }

    private fun comparatorForSize(): Comparator<File> {
        return Comparator { lhs, rhs -> (lhs.length() - rhs.length()).toInt() }
    }


}
